src/elli_static.erl

```diff
@@ -110,3 +110,43 @@ maybe_file(Req, Prefix, Dir) ->
         _ ->
             nothing
     end.
+
+
+%% OTP_RELEASE macro was introduced in 21, `filename:safe_relative_path/1' in
+%% 19.3, so the code below is safe
+-ifdef(OTP_RELEASE).
+  -ifdef(?OTP_RELEASE >= 21).
```

Comment:
I don't think I've seen this sort of `ifdef` before, but it appears to work as desired.

Reply:
My preference would be to defer to `filename:safe_relative_path/1` if available.  However, checking this at runtime (via `module_info(exports)`) caused lots of coverall coverage complaints. 

Xref and dialyzer are run under the test profile, so in-module eunit tests ended up failing xref right out of the gate (`elli_static:safe_relative_path_test/0 is unused export`).  Doh!

 I didn't want to introduce a new module, but I suppose I could create something like `elli_static_utils`, place `safe_relative_path/1` in there, and test independently of the OTP version.  The utils module would be a cleaner approach IMO.

Reply:
:+1: to a utils module, if you're up for it.